src/main/java/uk/gov/ida/eventemitter/LocalEncrypter.java

```diff
@@ -0,0 +1,46 @@
+package uk.gov.ida.eventemitter;
+
+import com.google.inject.Inject;
+import org.apache.commons.codec.binary.Base64;
+
+import javax.crypto.Cipher;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+import java.io.ByteArrayOutputStream;
+import java.io.ObjectOutputStream;
+
+public class LocalEncrypter implements Encrypter {
+
+    private final String key;
+    private final String initVector;
+
+    @Inject
+    public LocalEncrypter(final String key, final String initVector) {
+        this.key = key;
+        this.initVector = initVector;
+    }
+
+    @Override
+    public String encrypt(final Event event) {
+        try {
+            final IvParameterSpec iv = new IvParameterSpec(initVector.getBytes("UTF-8"));
+            final SecretKeySpec skeySpec = new SecretKeySpec(key.getBytes("UTF-8"), "AES");
+            final Cipher cipher = Cipher.getInstance("AES/CBC/PKCS5PADDING");
+            cipher.init(Cipher.ENCRYPT_MODE, skeySpec, iv);
+
+            final byte[] encryptedEvent;
+
+            try (ByteArrayOutputStream baos = new ByteArrayOutputStream();
+                ObjectOutputStream oos = new ObjectOutputStream(baos)) {
+                oos.writeObject(event);
+                oos.flush();
+                encryptedEvent = cipher.doFinal(baos.toByteArray());
+            }
+
+            return Base64.encodeBase64String(encryptedEvent);
+        } catch (Exception e) {
+            System.err.println(String.format("Failed to encrypt an event (Event ID: %s). Error message: %s", event.getEventId(), e.getMessage()));
+        }
+        return null;
```

Comment:
Is this the behaviour that we want? If we fail to decrypt, shouldn't things crash and burn (or handle the error somehow), rather than continuing on with null things?

Reply:
I have removed the try and catch statement. It is best to let the Local Encrypter throw any exception to the caller and let it decide what action it should take when receiving the exception.